libtransmission/rpc-server.cc

```diff
@@ -686,6 +741,12 @@ void start_server(tr_rpc_server* server)
     }
     else
```

Comment:
If this branch is not taken (i.e. if `!success` is true), `m_ctx` is leaking. You should probably move `m_ctx` declaration here, will also mean one less `#ifdef` since the two will be merged together.

Reply:
The leak here has not been addressed, it's still happening if `!success` is true.

Moreover, as I look closer I see that `evhttp_set_bevcb()` call doesn't really transfer ownership of the `ctx` object to `httpd`, and neither does new SSL object created by `SSL_new()` take ownership of it (if the callback is even called, which may never happen). Both expect the pointer to stay valid while corresponding objects are alive, but they're not responible for freeing it when it's no longer needed. This means that you need to do this yourself, i.e. `ctx` should probably be a member variable of `tr_rpc_server`.

Reply:
added `ctx` to `tr_rpc_server`